### PR TITLE
Sundials 6.x support

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

**Comment by jschueller** [comment]
hi @PeterMeisrimelModelon, @modelonrobinandersson, do you mind taking a look ?

**Comment by modelonrobinandersson** [comment]
@jschueller I will try to run the tests this week with your branch, can you add a changelog entry?

**Comment by modelonrobinandersson** [comment]
> Also, what's the lowest version of sundials we want to support here ? it seems to go as far as 2.x (2015) maybe assuming sundials>=4.x is more reasonable ?

Sundials 2.x is still required, lower than that however is not necessary as far as I know.

File: src/lib/sundials_includes.pxd [hunk 1]
```diff
@@ -369,13 +416,19 @@ IF SUNDIALS_VERSION >= (3,0,0):
...
+        ELSE:
+            int CVDlsSetLinearSolver(void *cvode_mem, SUNLinearSolver LS, SUNMatrix A)
+            int CVDlsSetJacFn(void *cvode_mem, CVDlsDenseJacFn djac)
     cdef extern from "cvodes/cvodes_spils.h":
-        int CVSpilsSetLinearSolver(void *cvode_mem, SUNLinearSolver LS)
         ctypedef int (*CVSpilsJacTimesSetupFn)(realtype t, N_Vector y, N_Vector fy, void *user_data)
-        int CVSpilsSetJacTimes(void *cvode_mem, CVSpilsJacTimesSetupFn jtsetup, CVSpilsJacTimesVecFn jtimes)
-
+        IF SUNDIALS_VERSION >= (4,0,0):
+            int CVodeSetJacTimes(void *cvode_mem, CVSpilsJacTimesSetupFn jtsetup, CVSpilsJacTimesVecFn jtimes)
```
Comment (PeterMeisrimelModelon) [inline src/lib/sundials_includes.pxd:428]: Should there be an extra line 

int CVSpilsSetLinearSolver(void *cvode_mem, SUNLinearSolver LS)

or is this no longer required here?

Comment (jschueller) [reply src/lib/sundials_includes.pxd:428]: It is only used for sundials=3.x

**Pull request merged**